editmsg.c

```diff
@@ -66,13 +66,14 @@ static int ev_message(enum EvMessage action, struct Mailbox *m, struct Email *e)
   struct Buffer *fname = mutt_buffer_pool_get();
   mutt_buffer_mktemp(fname);
 
-  enum MailboxType otype = C_MboxType;
-  C_MboxType = MUTT_MBOX;
+  unsigned char c_mbox_type = cs_subset_enum(NeoMutt->sub, "mbox_type");
+  enum MailboxType otype = c_mbox_type;
+  c_mbox_type = MUTT_MBOX;
```

Comment:
So my `mbox_type` is maildir, but we need `mbox` here to create a single temporary file. However, you're not overriding the global value `C_MboxType` of the `mbox_type` variable anymore, but just the local copy `c_mbox_type`. Then `mx_mbox_open` fetches `mbox_type` again, finds `maildir`, and the temporary file becomes a temporary directory with `cur`, `tmp`, `new`. At line 111 we stat it thinking it's a file, but because it's a directory we're confused and boom.

Reply:
I fear this is a subtle bug that might have crept in in several places, not easily spotted by static analysis :(

Reply:
Ah, I see.  Thanks.
Fixed in 7985e4d043.

Reply:
> I fear this is a subtle bug that might have crept in in several places,

You're probably right...

> not easily spotted by static analysis

Actually, it should be possible.

To make the refactoring quicker, I generated example `variable = helper_function()` for every config variable.
I **meant** to make them all `const` variables to avoid situations like this.
Looks like I missed some.

I'll review the constness of all of them,
and re-review the ~180 places where we set variables
(to make sure noone relies on an old variable).

Reply:
>  I **meant** to make them all `const` variables to avoid situations like this.

ah yes, that makes sense!